File: leetcode/409.LongestPalindrome.go
package leetcode

import "math"

func longestPalindrome(s string) int {
	mappings := make(map[string]int)

	for _, char := range s {
		_, ok := mappings[string(char)]
		if !ok {
			mappings[string(char)] = 0
		}
		mappings[string(char)] += 1
	}

	count := 0
	for _, m := range mappings {
		count += int(math.Floor(float64(m/2))) * 2

		if count%2 == 0 && m%2 == 1 {
			count += 1
		}
	}

	return count
}

//result := longestPalindrome("abccccdd")
//
//fmt.Printf("%+v", result)

File: leetcode/409.LongestPalindrome.ts
function longestPalindrome(s: string): number {
  const maps = s.split('').reduce((acc, v) => {
    acc[v] = (acc[v] ?? 0) + 1
    return acc
  }, {} as Record<string, number>)

  let count = 0
  for (const key in maps) {
    count += Math.floor(maps[key] / 2) * 2

    if (count % 2 === 0 && maps[key] % 2 === 1) {
      count += 1
    }
  }

  return count
}
console.log(longestPalindrome('abccccdd'))
